apache/superset
# chore(explore): Update chart save to use API endpoints

### SUMMARY

When a user saves a chart in Explore, the client sends one `POST /explore` request, then the server creates/updates the chart and optionally adds it to a new/existing dashboard as indicated.  This PR updates Explore's `SaveModal` to instead use v1 API endpoints to accomplish the same.  Here's the updated sequence of API calls:
- If user is creating a new dashboard: `POST /api/v1/dashboard/`
- If user is adding to a new dashboard: `PUT /api/v1/dashboard/{dashboard_id}`
- If user is saving as a new chart: `POST /api/v1/chart/`
  - If adding to new/existing dashboard: `{ dashboards: [..., dashboard_id] }`
- If user is overwriting an existing chart: `PUT /api/v1/chart/{chart_id}`
  - If adding to new/existing dashboard: `{ dashboards: [..., dashboardId] }`

In addition, this PR makes the Save and Save and Go to Dashboard buttons no longer trigger a page refresh.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS

Try performing the following tasks with and without proposed update and ensure that interface and user experience are identical (with the exception of there no longer being page refreshes):
- Create new chart from scratch, save as new chart, don't add to dashboard
- Create new chart from scratch, save as new chart, add to existing dashboard
- Create new chart from scratch, save as new chart, add to new dashboard
- Edit existing chart, overwrite existing chart, don't add to dashboard
- Edit existing chart, overwrite existing chart, add to existing dashboard
- Edit existing chart, overwrite existing chart, add to new dashboard
- Edit existing chart, save as new chart, don't add to dashboard
- Edit existing chart, save as new chart, add to existing dashboard
- Edit existing chart, save as new chart, add to new dashboard

### ADDITIONAL INFORMATION
